snowflake-api/src/connection.rs

```diff
@@ -77,9 +77,29 @@ pub struct Connection {
 
 impl Connection {
     pub fn new() -> Result<Self, ConnectionError> {
+        let client = Self::default_client_builder()?;
+
+        Ok(Self::new_with_middware(client.build()))
+    }
+
+    /// Allow a user to provide their own middleware
+    ///
+    /// Users can provide their own middleware to the connection like this:
+    ///
+    /// ```ignore
+    /// let mut builder = Connection::default_client_builder();
+    /// ...<modify the builder here>
+    /// let connection = Connection::new_with_middware(client.build());
+    /// ```
```

Comment:
```suggestion
    /// use snowflake_api::connection::Connection;
    /// let mut client = Connection::default_client_builder();
    ///  // modify the client builder here
    /// let connection = Connection::new_with_middware(client.unwrap().build());
```

Reply:
If you care about running the doctest (just to validate imports) we can add back the markdown block ```, otherwise I'm fine merging as is :)

Reply:
@wseaton sorry, forgot to include backticks, could you commit them back so it's actually a doctest?

Reply:
Sure!